modules/@apostrophecms/page/index.js

```diff
@@ -3233,24 +3233,35 @@ database.`);
           });
         });
       },
-      async inferLastTargetIdAndPosition(doc) {
+      async inferLastTargetIdAndPosition(doc, { publishedTargetsOnly = false } = {}) {
         const parentPath = self.getParentPath(doc);
         const parentAposDocId = parentPath.split('/').pop();
         const parentId = doc.aposLocale
           ? `${parentAposDocId}:${doc.aposLocale}`
           : parentAposDocId;
+        if (publishedTargetsOnly) {
+          const parent = await self.apos.doc.db.findOne({ _id: parentId });
+          if (!parent?.lastPublishedAt) {
+            throw self.apos.error('notfound');
```

Comment:
We have a strict policy against publishing children of unpublished parents and the UI enforces that etc., so all we need to do is sanity check here. It works out for doc template library because doc templates are children of the home page, and parked pages are always published.